cmd/snap-update-ns/utils.go

```diff
@@ -202,6 +204,63 @@ func secureMkFile(fd int, segments []string, i int, perm os.FileMode, uid sys.Us
 	return nil
 }
 
+// secureMkSymlink creates a symlink at i-th entry of absolute path represented by
+// segments. This function is meant to be used to create the leaf symlink.
+// Existing and identical symlinks are reused without errors.
+func secureMkSymlink(fd int, segments []string, i int, oldname string) error {
+	logger.Debugf("secure-mk-symlink %d %q %d %q", fd, segments, i, oldname)
+	segment := segments[i]
+	var err error
+
+	// Open the final path segment as a file. Try to create the file (so that
+	// we know if we need to chown it) but fall back to just opening an
+	// existing one.
+
+	// TODO: don't write links outside of tmpfs or $SNAP_{,USER_}{DATA,COMMON}
+	err = sysSymlinkat(oldname, fd, segment)
+	if err != nil {
+		switch err {
+		case syscall.EEXIST:
+			var objFd int
+			const O_PATH = 010000000
```

Comment:
Pity it's not a part of Go stdlib.

Reply:
Yeah, one more thing that's annoyingly missing.